strongbox-storage/strongbox-storage-api/src/main/java/org/carlspring/strongbox/services/ArtifactManagementService.java

```diff
@@ -141,12 +151,16 @@ public void releaseLock(URI pathUri)
 
     public void acquireLock(URI pathUri)
     {
-        Lock lock = Optional.ofNullable(pathUriMap.putIfAbsent(pathUri, lock = new ReentrantLock())).orElse(lock);
+        Lock lock = Optional.ofNullable(pathUriMap.putIfAbsent(pathUri,
+                                                               lock = hazelcastInstance.getLock(pathUri.toString())))
```

Comment:
@sbespalov : Do we still need to be keeping the URI-s in the map, or should we let Hazelcast handle this?

Reply:
yeah, we don't need this Map with Hazelcast because Hazelcats will be this Map :)
but we should think how to invalidate `ILock` objects like this was done in `ArtifactManagementService.releaseLock` to avoid memory leaks. I think we should use `ILock.destroy()` somehow, maybe there is some examples of how to do this correctly in concurrent environment.